packages/nuxt/src/pages/utils.ts

```diff
@@ -64,17 +64,14 @@ export async function resolvePagesRoutes (): Promise<NuxtPage[]> {
     shouldUseServerComponents: !!nuxt.options.experimental.componentIslands,
   })
 
-  const pages = uniqueBy(allRoutes, 'path')
-
   const shouldAugment = nuxt.options.experimental.scanPageMeta || nuxt.options.experimental.typedPages
 
+  const pages = uniqueBy(allRoutes, 'path')
+
+  await nuxt.callHook('pages:extend', pages)
   if (shouldAugment) {
-    const augmentedPages = await augmentPages(pages, nuxt.vfs)
-    await nuxt.callHook('pages:extend', pages)
-    await augmentPages(pages, nuxt.vfs, augmentedPages)
-    augmentedPages.clear()
-  } else {
-    await nuxt.callHook('pages:extend', pages)
+    await augmentPages(pages, nuxt.vfs)
+    await nuxt.callHook('pages:resolved', pages)
   }
```

Comment:
```suggestion
  }
  await nuxt.callHook('pages:resolved', pages)
```

maybe right before returning pages ?